### ISSUE TITLE
[bitnami/rabbitmq] HTTP access denied: user 'admin' - invalid credentials

### ISSUE THREAD

**Issue by evgmoskalenko** [bitnami/charts#4033]
**Which chart**:

* **Chart:** [bitnami/rabbitmq]
* **version:** 7.6.8
* **appVersion:** 3.8.9

**Describe the bug**

Hello, after deploying the chart, I try to login via UI http://localhost:15672/ and look error in logs:

```
│ 2020-10-16 06:08:12.154 [warning] <0.929.0> HTTP access denied: user 'admin' - invalid credentials│
│ 2020-10-16 06:08:23.785 [warning] <0.936.0> HTTP access denied: user 'provisioner' - Not management user|
│ 2020-10-16 07:21:06.618 [warning] <0.1851.1> HTTP access denied: user 'user' - invalid credentials|
```

**To Reproduce**
1. Deploy chart 

```
helm upgrade --install -f ./values.yaml --set auth.username=admin,auth.password=G84i8W1GKfilq4nW,auth.erlangCookie=kjhwekrhkwe8r8yiuhwqekjrhkjy rabbitmq bitnami/rabbitmq -n infrastructure --wait --timeout 300s --debug
```

with values:

```
auth:
  username: admin
  password: G84i8W1GKfilq4nW

extraEnvVars:
  - name: RABBITMQ_ERLANG_COOKIE
    value: kjhwekrhkwe8r8yiuhwqekjrhkjy
  - name: RABBITMQ_USERNAME
    value: admin
  - name: RABBITMQ_PASSWORD
    value: G84i8W1GKfilq4nW

extraConfiguration: |-
  default_vhost = /
  #disk_free_limit.absolute = 50MB
  load_definitions = /app/load_definition.json

loadDefinition:
  enabled: true
  existingSecret: "rabbitmq-load-definition"

replicaCount: 3
```

and 'load_definition.json' as secrets:

```
apiVersion: v1
kind: Secret
metadata:
  name: rabbitmq-load-definition
  namespace: infrastructure
type: Opaque
stringData:
  load_definition.json: |-
    {
      "vhosts": [
        {
          "name": "/"
        },
        {
          "name": "/my_id"
        }
      ],
      "policies": [
        {
          "name": "ha-all",
          "pattern": ".*",
          "vhost": "/my_id",
          "definition": {
            "ha-mode": "all",
            "ha-sync-mode": "automatic",
            "ha-sync-batch-size": 1
          }
        },
      ],
      "users": [
        {
          "name": "provisioner",
          "password": "G84i8W1GKfilq4nW",
          "tags": "admin"
        },
        {
          "name": "rabbitmq_ro",
          "password": "G84i8W1GKfilq4nW",
          "tags": "readonly"
        },
        {
          "name": "rabbitmq_rw",
          "password": "2WsrCKVecE7rg0J0",
          "tags": "readwrite"
        },
      ],
      "permissions": [
        {
          "user": "provisioner",
          "vhost": "/",
          "configure": ".*",
          "write": ".*",
          "read": ".*"
        },
        {
          "user": "rabbitmq_ro",
          "vhost": "/my_id",
          "configure": ".*",
          "write": ".*",
          "read": ".*"
        },
        {
          "user": "rabbitmq_rw",
          "vhost": "/my_id",
          "configure": ".*",
          "write": ".*",
          "read": ".*"
        },
      ],
      "queues": [
        {
          "name":"foo.bar.my_id.default",
          "vhost":"/my_id",
          "durable":true,
          "auto_delete":false,
          "arguments":{}
        }
      ],
      "exchanges": [
        {
          "name":"foo.bar",
          "vhost":"/my_id",
          "type":"topic",
          "durable":true,
          "auto_delete":false,
          "internal":false,
          "arguments":{}
        },
        {
          "name":"foo.bar",
          "vhost":"/my_id",
          "type":"topic",
          "durable":true,
          "auto_delete":false,
          "internal":false,
          "arguments":{}
        }
      ],
      "bindings": [
        {
          "source":"foo.bar",
          "vhost":"/my_id",
          "destination":"foo.bar.my_id.default",
          "destination_type":"queue",
          "routing_key":"foo.bar.my_id.default",
          "arguments":{}
        }
      ]
    }
```

Logs:

```
│ 2020-10-16 06:06:45.872 [info] <0.579.0> Importing concurrently 3 users...│
│ 2020-10-16 06:06:45.882 [info] <0.576.0> Successfully changed password for user 'provisioner'|
│ 2020-10-16 06:06:45.882 [info] <0.577.0> Successfully changed password for user 'rabbitmq_ro'|
│ 2020-10-16 06:06:45.884 [info] <0.576.0> Successfully set user tags for user 'provisioner' to [admin]|
│ 2020-10-16 06:06:45.884 [info] <0.577.0> Successfully set user tags for user 'rabbitmq_ro' to [readonly]|
│ 2020-10-16 06:06:45.898 [info] <0.576.0> Successfully changed password for user 'rabbitmq_rw'|
│ 2020-10-16 06:06:45.900 [info] <0.576.0> Successfully set user tags for user 'rabbitmq_rw' to [readwrite]|
│ 2020-10-16 06:06:45.900 [info] <0.579.0> Importing concurrently 2 vhosts...│
│ 2020-10-16 06:06:45.901 [info] <0.579.0> Importing concurrently 3 permissions...│
│ 2020-10-16 06:06:45.903 [info] <0.576.0> Successfully set permissions for 'provisioner' in virtual host '/' to '.*', '.*', '.*'|
│ 2020-10-16 06:06:45.903 [info] <0.577.0> Successfully set permissions for 'rabbitmq_ro' in virtual host '/my_id' to '.*', '.*', '.*'|
│ 2020-10-16 06:06:45.905 [info] <0.576.0> Successfully set permissions for 'rabbitmq_rw' in virtual host '/my_id' to '.*', '.*', '.*'|
│ 2020-10-16 06:06:45.905 [info] <0.579.0> Importing sequentially 1 policies...│
│ 2020-10-16 06:06:45.907 [info] <0.579.0> Importing concurrently 1 queues...│
│ 2020-10-16 06:06:45.908 [info] <0.579.0> Importing concurrently 2 exchanges...│
│ 2020-10-16 06:06:45.912 [info] <0.579.0> Importing concurrently 1 bindings...│
│ 2020-10-16 06:06:45.913 [info] <0.579.0> Ready to start client connection listeners|
│ 2020-10-16 06:06:45.916 [info] <0.868.0> started TCP listener on [::]:5672|
```

2. Try to login with credentials: admin, provisioner, or user with password bitnami

Error logs:

```
│ 2020-10-16 06:08:12.154 [warning] <0.929.0> HTTP access denied: user 'admin' - invalid credentials│
│ 2020-10-16 06:08:23.785 [warning] <0.936.0> HTTP access denied: user 'provisioner' - Not management user|
│ 2020-10-16 07:21:06.618 [warning] <0.1851.1> HTTP access denied: user 'user' - invalid credentials|
```

**Expected behavior**

I expect to be able to register under the created users

**Version of Helm and Kubernetes**:

*EKS version:* v1.18.8-eks-7c9bda

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.3.4", GitCommit:"a61ce5633af99708171414353ed49547cf05013d", GitTreeState:"dirty", GoVersion:"go1.15.2"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T21:51:49Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.8-eks-7c9bda", GitCommit:"7c9bda52c425d0d56d7b93f1377a826b4132c05c", GitTreeState:"clean", BuildDate:"2020-08-28T23:04:33Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

Maybe I'm not configuring correctly - "load_definition.json", but it seems looks like a good.. :-)

**Comment by evgmoskalenko** [comment]
I will try again these settings:

```
    "users": [
        {
          "name": "admin",
          "password": "G84i8W1GKfilq4nW",
          "tags": "administrator"
        },
        {
          "name": "provisioner",
          "password": "G84i8W1GKfilq4nW",
          "tags": "management"
        },
    ]
```

and 

```
      "permissions": [
        {
          "user": admin,
          "vhost": "/",
          "configure": ".*",
          "read": ".*",
          "write": ".*"
        },
        {
          "user": provisioner,
          "vhost": "/",
          "configure": ".*",
          "read": ".*",
          "write": ".*"
        }
      ],
```

But it would be correct to specify an example in the documentation, and test if the chart with these settings.. :-)

**Comment by carrodher** [comment]
Thanks for creating this issue, are you facing this issue on a fresh installation or upgrading an existing deployment? Please take into account 

In the installation notes, you should see some instructions about how to retrieve the different passwords, can you run those commands in order to check their values? You can obtain the installation notes by running
```
$ helm get notes rabbitmq
```

**Comment by evgmoskalenko** [comment]
Yes, I installed a new release with the disable pvc:

```
persistence:
  enabled: false
  selector: {}
  accessMode: ReadWriteOnce
```

After my installation I check my password via commands:

```
echo "Username      : admin"
echo "Password      : $(kubectl get secret --namespace infrastructure rabbitmq -o jsonpath="{.data.rabbitmq-password}" | base64 --decode)"
echo "ErLang Cookie : $(kubectl get secret --namespace infrastructure rabbitmq -o jsonpath="{.data.rabbitmq-erlang-cookie}" | base64 --decode)"
```

The passwords were specified as I set..

**Comment by evgmoskalenko** [comment]
I assume that this was due to the fact that I did not add the administrator and the provisioner users with the correct tag to the config:

```
    "users": [
        {
          "name": "admin",
          "password": "G84i8W1GKfilq4nW",
          "tags": "administrator"
        },
        {
          "name": "provisioner",
          "password": "G84i8W1GKfilq4nW",
          "tags": "management"
        },
    ]
```

Now my config looks like correct:

```
apiVersion: v1
kind: Secret
metadata:
  name: rabbitmq-load-definition
  namespace: infrastructure
type: Opaque
stringData:
  load_definition.json: |-
    {
      "global_parameters": [
        {
            "name": "cluster_name",
            "value": "rabbitmq"
        }
      ],
      "users": [
        {
          "name": "admin",
          "password": "G84i8W1GKfilq4nW",
          "tags": "administrator"
        },
        {
          "name": "provisioner",
          "password": "G84i8W1GKfilq4nW",
          "tags": "management"
        },
        {
          "name": "myidvhosts_ro",
          "password": "G84i8W1GKfilq4nW",
          "tags": "readonly"
        },
        {
          "name": "myidvhosts_rw",
          "password": "2WsrCKVecE7rg0J0",
          "tags": "readwrite"
        }
      ],
      "vhosts": [
        {
          "name": "/"
        },
        {
          "name": "/myidvhosts"
        }
      ],
      "permissions": [
        {
          "user": "admin",
          "vhost": "/",
          "configure": ".*",
          "read": ".*",
          "write": ".*"
        },
        {
          "user": "provisioner",
          "vhost": "/",
          "configure": ".*",
          "read": ".*",
          "write": ".*"
        },
        {
          "user": "myidvhosts_ro",
          "vhost": "/myidvhosts",
          "configure": ".*",
          "write": ".*",
          "read": ".*"
        },
        {
          "user": "myidvhosts_rw",
          "vhost": "/myidvhosts",
          "configure": ".*",
          "write": ".*",
          "read": ".*"
        }
      ],
      "policies": [
        {
          "name": "ha-all",
          "pattern": ".*",
          "vhost": "/myidvhosts",
          "definition": {
            "ha-mode": "all",
            "ha-sync-mode": "automatic",
            "ha-sync-batch-size": 1
          }
        }
      ],
      "queues": [
        {
          "name":"foo.bar.myidvhosts.default",
          "vhost":"/myidvhosts",
          "durable":true,
          "auto_delete":false,
          "arguments":{}
        }
      ],
      "exchanges": [
        {
          "name":"foo.bar",
          "vhost":"/myidvhosts",
          "type":"topic",
          "durable":true,
          "auto_delete":false,
          "internal":false,
          "arguments":{}
        },
        {
          "name":"foo.bar",
          "vhost":"/myidvhosts",
          "type":"topic",
          "durable":true,
          "auto_delete":false,
          "internal":false,
          "arguments":{}
        }
      ],
      "bindings": [
        {
          "source":"foo.bar",
          "vhost":"/myidvhosts",
          "destination":"foo.bar.myidvhosts.default",
          "destination_type":"queue",
          "routing_key":"foo.bar.myidvhosts.default",
          "arguments":{}
        }
      ]
    }
```

But now I can't test my theory yet, because I faced other problems

**Comment by evgmoskalenko** [comment]
By the way, with such a configuration, even if the PVC is used, the passwords change OK. I think it would be great to make such a config through the configmap in the chart and parameterize it from the values, and not deploy it separately from the chart

**Comment by carrodher** [comment]
We are used to implementing suggestions from users in order to improve the Bitnami Helm Charts, if you think there is a better way, please, feel free to create a PR implementing your suggested approach and we will be happy to review it!

**Comment by Gefuxing** [comment]
I also encountered the same problem. The password of the account I created was correct, but I failed to log in. Has the author tried this problem？

`2020-11-13 03:03:41.401 [warning] <0.1565.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:42.907 [warning] <0.1568.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:44.258 [warning] <0.1570.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:45.389 [warning] <0.1571.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:46.951 [warning] <0.1572.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:48.392 [warning] <0.1573.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:49.618 [warning] <0.1575.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:50.776 [warning] <0.1576.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:52.344 [warning] <0.1577.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:53.660 [warning] <0.1581.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:54.989 [warning] <0.1582.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:56.438 [warning] <0.1587.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:03:57.943 [warning] <0.1588.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:04:01.552 [warning] <0.1591.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:15:38.134 [warning] <0.2070.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:15:39.826 [warning] <0.2072.0> HTTP access denied: user 'user' - invalid credentials
2020-11-13 03:15:41.113 [warning] <0.2073.0> HTTP access denied: user 'user' - invalid credentials
`

**Comment by carrodher** [comment]
Are you facing this issue in a fresh installation or during an upgrade? Please, take into account if you are upgrading an existing deployment you need to set the existing passwords, if not there is going to be an inconsistency between the password set in the first installation and the one generated during the upgrade, please, take a look at https://github.com/bitnami/charts/tree/master/bitnami/rabbitmq#upgrading

**Comment by Gefuxing** [comment]
I'm just following the tutorial to deploy a new MQ cluster with the same configuration as the person above  enabled, Load Definitions, extraSecrets:
load-definition:
load_definition.json: |
{
"vhosts": [
{
"name": "/"
}
]
}
loadDefinition:
enabled: true
existingSecret: load-definition
extraConfiguration: |
load_definitions = /app/load_definition.json
------------------------------------------------------------------
After setting up the connection, the login failed,Is there a solution？

**Comment by juan131** [comment]
Hi @Gefuxing 

> The password of the account I created was correct

Are you using the user/password following the instructions obtained in the NOTES displayed after installing the chart? Or are you creating a different account?

> I'm just following the tutorial to deploy a new MQ cluster with the same configuration as the person above enabled,

Could you share the command you used to create the "load-definition" secret?
Did you modify any other parameter in your **values.yaml** (apart from the ones you already shared)?

**Comment by marcosbc** [comment]
Hi all, this is all expected because when you specify `load-definitions`, the password for the admin user cannot be changed in a secure way, so that step is skipped during the initialization.

Recently we implemented support for a new `RABBITMQ_SECURE_PASSWORD` environment variable to workaround this. If you set `RABBITMQ_SECURE_PASSWORD=no` via the chart's `extraEnvVars` parameters you can configure the initialization to set the password in a plaintext file. While it's not a secure way, at least it will allow you to set a password with `load-definitions` enabled. Refer to the [README](https://github.com/bitnami/bitnami-docker-rabbitmq#node-and-cluster-configuration) for more configurations.

**Comment by justinmchase** [comment]
I'm having this same issue. I just got the latest chart edited these values only:
```yaml
auth:
  username: user
  password: test
rbac:
  create: false
networkPolicy:
  enabled: true
  allowExternal: true
```

Deployed successfully but cannot log into the manager:

![Screen Shot 2021-01-19 at 4 42 34 PM](https://user-images.githubusercontent.com/10974/105103100-9573cd80-5a75-11eb-86c7-ed83087a408f.png)

From logs
```
2021-01-19 22:41:29.974 [info] <0.531.0> Resetting node maintenance status
2021-01-19 22:42:21.715 [warning] <0.873.0> HTTP access denied: user 'user' - invalid credentials
2021-01-19 22:42:25.914 [warning] <0.878.0> HTTP access denied: user 'user' - invalid credentials
```

**Comment by juan131** [comment]
Hi @justinmchase 

I was unable to reproduce the issue. 

Did you install the chart on a "clean" environment? I mean, with no existing PVCs from previous installations. See [this section](https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#persistence-volumes-pvs-retained-from-previous-releases) of the Troubleshooting guide for more information.

Just curious: Why did you disable RBAC? Does your cluster uses it?

**Comment by justinmchase** [comment]
Yeah my cluster forbids it sadly, its a hosted cluster with multiple tenants and we are forbidden from creating service accounts and most roles. Do you think that could be related to the issue?

**Comment by justinmchase** [comment]
I wiped everything out of my local kubernetes and reinstalled just rabbit and it seemed to work. I'm not sure because I wasn't seeing errors about it but it seems like the 8gi volume claim may have been too big for what I had setup by default and it was failing to mount the disk. After repairing my volumes a bit, manually, it now works as expected. Sorry for false alarm.

**Comment by juan131** [comment]
Great @justinmchase !!
I'm very glad that you could workaround the issues by repairing your volumes!

> Yeah my cluster forbids it sadly, its a hosted cluster with multiple tenants and we are forbidden from creating service accounts and most roles. Do you think that could be related to the issue?

This chart uses the [RabbitMQ Peer Discovery Kubernetes](https://github.com/rabbitmq/rabbitmq-peer-discovery-k8s) that requires to make some basic consults to K8s API to discover other peers. If you have RBAC enabled in your cluster, you many find issues when setting `rbac.create=false`.

**Comment by justinmchase** [comment]
@juan131 It needs to find the peers for clustering correct?

Which parts of the API does it need access to, the pods api I'm guessing? I am actually able to create such a Role and RoleBinding (I've done it for other pods) but I have to do it outside of this chart as it requires a User account to create them and not a ServiceRole which is what I'm using to deploy from the CI server.

Thanks for letting me know though, sadly dealing with multi-tenancy is fairly complicated.

**Comment by juan131** [comment]
Hi @justinmchase 

It consults the API for peers discovery. As you can see in the Role below, it just need permissions to "get" endpoints, and "create" events.

- https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/templates/role.yaml

**Comment by grahambunce** [comment]
FYI: Same problem for me too. In my case I have an installed version of rabbit via helm, uninstalled via helm, waited several days and then installed again. Cannot login via management console due to "invalid credentials".

When I get the secret via the instructions in the Helm notes, the secret matches what I am entering for the correct user (even tried copy/paste). Checked the secret in K8S dashboard - this matches expected too.

user and password is added on helm command line:

auth.username=admin,auth.password=admin

**Comment by datavistics** [comment]
I had this issue but was able to login after "Emptying Cache and Hard Reload". 

On Chrome/Brave:
1. Hit F12
2. Right Click the refresh button
3. Select Empty Cache and Hard Reload

**Comment by juan131** [comment]
Hi @grahambunce 

What you described looks like an issue with RabbitMQ data storing a different password from the one you have in the secret. That happens when you upgrade the RabbitMQ chart without passing the previous password.

This is properly explained in the Troubleshooting guide below:

- https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases

**Comment by justinmchase** [comment]
So I ended up solving this and I'm not sure what was going wrong but it seemed related to volumes. I had a "default volume" setup in my local cluster and it was out of disk space and so I think rabbit was failing to get a writable volume and was just failing at some level i wasn't able to diagnose.

So for me I ended up fixing this by wiping out everything on my local cluster, including volumes and setting up a new basic persistent volume as the default volume and then once I installed rabbit again it just worked right away.

**Comment by juan131** [comment]
Great @justinmchase !!! I'm very glad you were able to solve the authentication issues by using new volumes.

**Comment by anthonyvk** [comment]
I am facing this issue when using Hashicorp vault.

```
vault write rabbitmq/config/connection connection_uri="http://rabbitmq-svc:15672" username="admin" password="password"
Error writing data to rabbitmq/config/connection: Error making API request.

URL: PUT http://127.0.0.1:8200/v1/rabbitmq/config/connection
Code: 500. Errors:

* 1 error occurred:
        * failed to validate the connection: Error 401 (not_authorised): Login failed
```

RabbitMQ logs : 
`2021-03-17 08:02:26.249 [warning] <0.2243.0> HTTP access denied: user 'admin' - invalid credentials`

**Comment by juan131** [comment]
Hi @anthonyvk 

Sorry we missed your post. My apologizes

What did you exactly do to reproduce the issue? Could you share the exact commands you run?

**Comment by dev-comrade** [comment]
My solution:

Try to login with password "CHANGEME" :)

**Comment by marcosbc** [comment]
@dev-comrade That should already have been fixed as per #5939, I'd recommend to verify that you are using the latest chart version.

**Comment by carrodher** [comment]
This issue should already have been fixed at https://github.com/bitnami/charts/pull/5939, I'd recommend verifying that you are using the latest chart version.

If the issue persists, can you please create a new issue adding as much info as possible about your environment and how are you deploying the Helm chart?
